src/test/java/com/spotify/dns/DnsSrvResolversIT.java

```diff
@@ -46,7 +54,30 @@ public void setUp() throws Exception {
 
   @Test
   public void shouldReturnResultsForValidQuery() throws Exception {
-    assertThat(resolver.resolve("_spotify-client._tcp.sto.spotify.net").isEmpty(), is(false));
+    assertThat(resolver.resolve("_spotify-client._tcp.spotify.com").isEmpty(), is(false));
+  }
+
+  @Test
+  public void testCorrectSequenceOfNotifications() throws Exception {
+    ChangeNotifier<LookupResult> notifier = ChangeNotifiers.aggregate(
+        DnsSrvWatchers.newBuilder(resolver)
+            .polling(100, TimeUnit.MILLISECONDS)
+            .build().watch("_spotify-client._tcp.spotify.com"));
+
+    final List<String> changes = Collections.synchronizedList(Lists.<String>newArrayList());
+
+    notifier.setListener(new ChangeNotifier.Listener<LookupResult>() {
+      @Override
+      public void onChange(ChangeNotifier.ChangeNotification<LookupResult> changeNotification) {
+        Set<LookupResult> current = changeNotification.current();
+        if (!ChangeNotifiers.isInitialEmptyData(current)) {
+          changes.add(current.isEmpty() ? "empty" : "data");
+        }
+      }
+    }, true);
+    assertEquals(ImmutableList.of(), changes);
+    Thread.sleep(200);
```

Comment:
To make this somewhat less brittle, maybe it would be good to use https://github.com/awaitility/awaitility instead of a simple sleep?